apps/emqx_bridge/src/emqx_bridge_resource.erl

```diff
@@ -243,12 +243,18 @@ create_dry_run(Type, Conf0) ->
         {error, Reason} ->
             {error, Reason};
         {ok, ConfNew} ->
-            ParseConf = parse_confs(bin(Type), TmpPath, ConfNew),
-            Res = emqx_resource:create_dry_run_local(
-                bridge_to_resource_type(Type), ParseConf
-            ),
-            _ = maybe_clear_certs(TmpPath, ConfNew),
-            Res
+            try
+                ParseConf = parse_confs(bin(Type), TmpPath, ConfNew),
+                Res = emqx_resource:create_dry_run_local(
+                    bridge_to_resource_type(Type), ParseConf
+                ),
+                _ = maybe_clear_certs(TmpPath, ConfNew),
+                Res
+            catch
+                %% validation errors
+                throw:Reason ->
+                    {error, Reason}
```

Comment:
Maybe we should call `maybe_clear_certs` in a `finally` block to ensure it cleans up even in the failure case?

Reply:
Thanks, fixed.